modules/ref_fb_module/src/classifier_fb_impl.cpp

```diff
@@ -0,0 +1,360 @@
+    };
+}
+
+void ClassifierFbImpl::processEventPacket(const EventPacketPtr& packet)
+{
+    if (packet.getEventId() == event_packet_id::DATA_DESCRIPTOR_CHANGED)
+    {
+        DataDescriptorPtr inputDataDescriptor = packet.getParameters().get(event_packet_param::DATA_DESCRIPTOR);
+        DataDescriptorPtr inputDomainDataDescriptor = packet.getParameters().get(event_packet_param::DOMAIN_DATA_DESCRIPTOR);
+        processSignalDescriptorChanged(inputDataDescriptor, inputDomainDataDescriptor);
+    }
+}
+
+inline bool ClassifierFbImpl::timeInInterval(UInt startTime, UInt endTime) {
+    return (endTime - startTime) < timeMs(blockSize);
+}
+
+template <SampleType InputSampleType>
+void ClassifierFbImpl::processDataPacket(const DataPacketPtr& packet)
+{
+    using InputType = typename SampleTypeToType<InputSampleType>::Type;
+    using OutputType = Float;
+
+    // if packet does not have samples - ignore
+    if (packet.getSampleCount() == 0) {
+        return;
+    }
+
+    packets.push_back(packet);
+
+    size_t outputPackages = 0;
+    {
+        auto inputDomainData = static_cast<UInt*>(packet.getDomainPacket().getData());
+        UInt lastTime = inputDomainData[packet.getSampleCount() - 1];
+
+        // initialize members
+        if (packetStarted == UInt{}) {
+            packetStarted = inputDomainData[0];
+            sampleStarted = 0; 
+        }
+
+        outputPackages = (lastTime - packetStarted) / timeMs(blockSize);
+    }
+
+    if (!outputPackages)
+        return;
+
+    auto outputDomainPacket = DataPacket(outputDomainDataDescriptor, 1);
+    auto outputDomainData = static_cast<UInt*>(outputDomainPacket.getData());
+
+    DataPacketPtr outputPacket = DataPacketWithDomain(outputDomainPacket, outputDataDescriptor, 1);
+    OutputType* outputData = static_cast<OutputType*>(outputPacket.getData());
+
+    auto rangeSize = outputDataDescriptor.getDimensions()[0].getSize();
+    auto labels = outputDataDescriptor.getDimensions()[0].getLabels();
+    if (labels.getCount() == 0) return;
+
+    size_t packageVals = 0;
+    while (outputPackages && packets.size()) {
+        auto listPacket = packets.front();
+
+        auto sampleCnt = listPacket.getSampleCount();
+        auto inputData = static_cast<InputType*>(listPacket.getData());
+        auto inputDomainData = static_cast<UInt*>(listPacket.getDomainPacket().getData());
+       
+        // reset array for new package
+        if (packageVals == 0) {
+            memset(outputData, 0, rangeSize * sizeof(OutputType));
+        }
+
+        bool packetInEpoch = timeInInterval(packetStarted, inputDomainData[sampleCnt - 1]);
+
+        size_t sampleIdx = sampleStarted;
+        for (; sampleIdx < sampleCnt; sampleIdx++) {
+            // if there are values from another interval - check each value
+            if (!packetInEpoch) {
+                if (!timeInInterval(packetStarted, inputDomainData[sampleIdx]))
+                    break;
+            }
```

Comment:
Not sure we need to consider this here, but this loop will fail if the domain delta between two samples is greater than our block size (in ms).

Reply:
we will just get out from loop and will send empty package